core/client/fs/src/main/java/alluxio/client/block/BlockMasterClientQueue.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block;
+
+import alluxio.conf.PropertyKey;
+import alluxio.master.MasterClientContext;
+import alluxio.resource.DynamicResourcePool;
+import alluxio.resource.RoundRobinResourceQueue;
+import alluxio.util.ThreadFactoryUtils;
+
+import java.io.IOException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+/**
+ * This class manages {@link BlockMasterClient} resources in a queue.
+ * See {@link RoundRobinResourceQueue} for details of how resources are organized.
+ */
+public class BlockMasterClientQueue extends RoundRobinResourceQueue<BlockMasterClient> {
+
+  private static final int BLOCK_MASTER_CLIENT_POOL_GC_THREADPOOL_SIZE = 3;
+  private static final ScheduledExecutorService GC_EXECUTOR =
+          new ScheduledThreadPoolExecutor(BLOCK_MASTER_CLIENT_POOL_GC_THREADPOOL_SIZE,
+                  ThreadFactoryUtils.build("BlockMasterClientQueueGcThreads-%d", true));
+
+  private final MasterClientContext mMasterClientContext;
+
+  /**
+   * Create a new {@link BlockMasterClientQueue}.
+   * @param ctx Client Context
+   */
+  public BlockMasterClientQueue(MasterClientContext ctx) {
+    super(DynamicResourcePool.Options.defaultOptions()
+        .setGcExecutor(GC_EXECUTOR)
+        .setGcIntervalMs(ctx.getClusterConf()
+            .getMs(PropertyKey.USER_BLOCK_MASTER_CLIENT_POOL_GC_INTERVAL_MS))
+        // todo: tentative parameter
+        .setMaxCapacity(64));
+    mMasterClientContext = ctx;
+  }
```

Comment:
The `maxCapacity` here is experimental. In principle I think we could use a much lower max capacity than the original 1024.